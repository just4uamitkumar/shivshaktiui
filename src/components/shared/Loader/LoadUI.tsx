import { CircularProgress, Stack } from "@mui/material";
import { Grid } from "@mui/material";
import React from "react";
import "./styles.scss";

const LoadUI: React.FC = () => {
  return (
    <Grid
      size={12}
      container
      className="noData-container"
      direction={"row"}
      alignItems={"center"}
      justifyContent={"center"}
    >
      <Stack>
        <CircularProgress color="success" variant="indeterminate" />
      </Stack>
    </Grid>
  );
};

export default LoadUI;
