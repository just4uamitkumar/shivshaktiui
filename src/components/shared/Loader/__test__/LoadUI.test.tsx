import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import LoadUI from "../index";

describe("LoadUI Component", () => {
  it("matches snapshot", () => {
    const { container } = render(<LoadUI />);
    expect(container).toMatchSnapshot();
  });

  it("should render without crashing", () => {
    render(<LoadUI />);
    expect(LoadUI).toBeDefined();
  });
  it("should render the loader with default size", () => {
    render(<LoadUI />);
    const loaderElement = screen.getByRole("progressbar");
    expect(loaderElement).toBeInTheDocument();
    expect(loaderElement).toHaveStyle({ width: "3rem", height: "3rem" });
  });
});
