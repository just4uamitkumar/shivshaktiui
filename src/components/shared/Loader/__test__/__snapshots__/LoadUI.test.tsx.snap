// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadUI Component > matches snapshot 1`] = `
<div>
  <div
    class="loader"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1wxecgq-MuiCircularProgress-root"
      role="progressbar"
      style="width: 3rem; height: 3rem;"
    >
      <svg
        class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-19t5dcl-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</div>
`;
